utils/custom_logger.py

```diff
@@ -2,53 +2,49 @@
 import os
 import time
 
-
-class CustomLogger:
-    """
-    Singleton instance for logging, use `CustomLogger().main_logger` to get the instance.
-    """
-
-    _instance = None
-    main_logger = None
-
-    def __new__(self):
-        if self._instance is None:
-            self._instance = super().__new__(self)
-            self._instance._setup()
-        return self._instance
-
-    def _setup(self):
-        if not os.path.exists("log"):
-            os.makedirs("log")
-
-        self.main_logger = logging.getLogger("mainlogger")
-        self.main_logger.setLevel(logging.DEBUG)
-
-        formatter_c = logging.Formatter(
-            "%(asctime)s  [\033[93m%(pathname)s:%(lineno)d\033[0m]  %(levelname)s: %(message)s"
-        )
-        formatter_f = logging.Formatter(
-            "%(asctime)s  [%(pathname)s:%(lineno)d]  %(levelname)s: %(message)s"
-        )
-
-        console_handler = logging.StreamHandler()
-        console_handler.setLevel(logging.INFO)
-        console_handler.setFormatter(formatter_c)
-        self.main_logger.addHandler(console_handler)
-
-        file_handler = logging.FileHandler(f"log/python-{time.time()}.log", mode="w")
-        file_handler.setLevel(logging.INFO)
-        file_handler.setFormatter(formatter_f)
-        self.main_logger.addHandler(file_handler)
-
-        file_handler_error = logging.FileHandler(f"log/error-{time.time()}.log", mode="w")
-        file_handler_error.setLevel(logging.ERROR)
-        file_handler_error.setFormatter(formatter_f)
-        self.main_logger.addHandler(file_handler_error)
-
-        file_handler_debug = logging.FileHandler(f"log/debug-{time.time()}.log", mode="w")
-        file_handler_debug.setLevel(logging.DEBUG)
-        file_handler_debug.setFormatter(formatter_f)
-        self.main_logger.addHandler(file_handler_debug)
-
-        return self.main_logger
+main_logger: logging.Logger = logging.getLogger('zpd')
+"""
+The main logger to be used by the program.
+"""
+
+if not os.path.exists("log"):
+    os.makedirs("log")
+
+# Allow all log levels
+main_logger.setLevel(logging.DEBUG)
+
+def setup():
+    global main_logger
+
+    formatter_c = logging.Formatter(
+        "%(asctime)s  [\033[93m%(pathname)s:%(lineno)d\033[0m]  %(levelname)s: %(message)s"
+    )
+    formatter_f = logging.Formatter(
+        "%(asctime)s  [%(pathname)s:%(lineno)d]  %(levelname)s: %(message)s"
+    )
+
+    # Add console log handler
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(logging.INFO)
+    console_handler.setFormatter(formatter_c)
+    main_logger.addHandler(console_handler)
+
+    # Add main file log handler
+    file_handler = logging.FileHandler(f"log/python-{time.time()}.log", mode="w")

```

Comment:
Instead of calling `time.time()` for each type of  log file, we could create a variable and use it for all of them. This way the files would have a more exact and consistent name.